### PR TITLE
Add validation for front end sign up form

### PR DESCRIPTION
**Pull request by EnixYY**

### CODE REVIEW

File: lunch-go-where/src/pages/Landing.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ const style = {
   border: "2px solid #000",
```
Comment (esskew213) [inline lunch-go-where/src/pages/Landing.js]: If you want you can consider removing this, do you think the modal looks a bit weird with a black border? Just a thought.

File: lunch-go-where/src/pages/Landing.js [hunk 2]
```diff
@@ -52,29 +55,51 @@ const Landing = () => {
...
-      const res = await apis.postSignUp({
-        name: name,
-        username: newUsername,
-        password: newPassword,
-      });
-      setName("");
-      setNewUsername("");
-      setNewPassword("");
+      await apis.login({ username: username, password: password });
+      setPassword("");
```
Comment (esskew213) [inline lunch-go-where/src/pages/Landing.js:74]: here it's good to reset the name and uesrname to "" as well?

**Review by esskew213 — DISMISSED** [review]

File: lunch-go-where/src/pages/Landing.js [hunk 3]
```diff
@@ -164,6 +191,15 @@ const Landing = () => {
               />
             </FormControl>
             <br />
+            {showErrorLabel ? (
+              <label>
+                Password is not valid. Please include a letter and a number
```
Comment (esskew213) [inline lunch-go-where/src/pages/Landing.js:196]: What happens if a user tries to type a space?

File: lunch-go-where/src/pages/Landing.js (hunk 1 above)
Comment (EnixYY) [reply lunch-go-where/src/pages/Landing.js]: Yes i think it looks better hehe thanksu! Made the changes :D

File: lunch-go-where/src/pages/Landing.js (hunk 2 above)
Comment (EnixYY) [reply lunch-go-where/src/pages/Landing.js:74]: Yes sorry i miss out for this have added it in

File: lunch-go-where/src/pages/Landing.js (hunk 3 above)
Comment (EnixYY) [reply lunch-go-where/src/pages/Landing.js:196]: Will work on this after card component is done. :D

**Review by the-observer07 — APPROVED** [review]

**Pull request merged**